src/test/java/com/amazon/opendistroforelasticsearch/knn/index/KNNESSettingsTestIT.java

```diff
@@ -87,5 +91,73 @@ public void testCreateIndexWithInvalidSpaceType() throws IOException {
             () -> createKnnIndex(INDEX_NAME, invalidSettings, createKnnIndexMapping(FIELD_NAME, 2)));
         assertThat(ex.getMessage(), containsString(String.format("Unsupported space type: %s", invalidSpaceType)));
     }
+
+    public void testUpdateIndexSetting() throws IOException {
+        Settings settings = Settings.builder()
+                .put("index.knn", true)
+                .put(KNNSettings.KNN_ALGO_PARAM_EF_SEARCH, 512)
+                .build();
+        createKnnIndex(INDEX_NAME, settings, createKnnIndexMapping(FIELD_NAME, 2));
+        assertEquals("512", getIndexSettingByName(INDEX_NAME, KNNSettings.KNN_ALGO_PARAM_EF_SEARCH));
+
+        updateIndexSettings(INDEX_NAME, Settings.builder().put(KNNSettings.KNN_ALGO_PARAM_EF_SEARCH, 400));
+        assertEquals("400", getIndexSettingByName(INDEX_NAME, KNNSettings.KNN_ALGO_PARAM_EF_SEARCH));
+
+        Exception ex = expectThrows(ResponseException.class,
+                () -> updateIndexSettings(INDEX_NAME,
+                        Settings.builder().put(KNNSettings.KNN_ALGO_PARAM_EF_SEARCH, 1)));
+        assertThat(ex.getMessage(),
+                containsString("Failed to parse value [1] for setting [index.knn.algo_param.ef_search] must be >= 2"));
+    }
+
+    public void testKNNStatsAfterUpdateIndexSetting() throws IOException {
```

Comment:
I think the purpose of this test case is to confirm that the Index is evicted from the cache after the settings are updated. That being said, I think the test should be renamed `testCacheRebuiltAfterUpdateIndexSettings`.

Further more, instead of comparing deltas of hit and miss count, you can check that the cache is empty with the `indices_in_cache` stat. Then, you can just
1. Create index
2. Index sample docs
3. Search index
4. Confirm index is present in cache
5. Update index settings
6. Confirm there are no indices in the cache.

Reply:
Cool.